app/uk/gov/hmrc/nisp/connectors/NispConnector.scala

```diff
@@ -64,22 +64,17 @@ trait NispConnector {
       keystoreResult match {
         case Some(data) =>
           MetricsService.keystoreHitCounter.inc()
-          Future.successful(Some(data))
+          Future.successful(data)
         case None =>
           MetricsService.keystoreMissCounter.inc()
-          connectToMicroservice[A](url, api) map {
+          connectToMicroservice[A](url, api) flatMap {
             case Success(data) =>
-              Some(cacheResult(data, api.toString))
+              Future.successful(cacheResult(data, api.toString))
             case Failure(ex) =>
               MetricsService.failedCounters(api).inc()
-              Logger.error(s"Backend microservice has returned no data for $api Response: $ex")
```

Comment:
I feel this flatMap could be refactored in to a onError somehow, as the future remains unchanged at the end.